packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -122,7 +128,48 @@ impl Abigen {
                            let provider = self.wallet.get_provider()?;
                            wallet.set_provider(provider.clone());
 
-                           Ok(Self { contract_id: self.contract_id.clone(), wallet: wallet })
+                           Ok(Self { contract_id: self.contract_id.clone(), wallet: wallet, logs_lookup: self.logs_lookup.clone() })
+                        }
+
+                        pub fn _logs_with_type<D: Tokenizable + Parameterize>(&self, receipts: &[Receipt]) -> Result<Vec<D>, SDKError> {
+                            let param_type = D::param_type();
+
+                            let target_ids: HashSet<u64> = self
+                                .logs_lookup
+                                .iter()
+                                .filter_map(|l| {
+                                    if l.param_type == param_type {
+                                        Some(l.log_id)
+                                    } else {
+                                        None
+                                    }
+                                })
+                                .collect();
+
+                            let decode_log_content = |content: &[u8]| {
+                                let dec =
+                                    ABIDecoder::decode_single(&D::param_type(), content)?;
+                                D::from_token(dec)
+                            };
+
+                            let decoded_logs: Vec<D> = receipts
```

Comment:
I'd risk needless allocations and try and make this more readable:
```rust
    let decoded_logs = receipts
        .iter()
        .filter_map(|r| match r {
            Receipt::LogData { rb, data, .. } if target_ids.contains(rb) => Some(data.clone()),
            Receipt::Log { ra, rb, .. } if target_ids.contains(rb) => {
                Some(ra.to_be_bytes().to_vec())
            }
            _ => None,
        })
        .map(|data| try_from_bytes(&data))
        .collect::<Result<Vec<_>, _>>()?;
```

Doing an `except` rather than handling `Result`s here is probably better, failing fast and all that... 

But then again it is hard to decide for our users what should kill their app.

Reply:
Nice. I was trying to think of a way to make this look better, guess I was to stingy with the allocations.